uqcsbot/join.py

```diff
@@ -0,0 +1,103 @@
+import discord
+from discord.ext import commands
+
+from uqcsbot.bot import UQCSBot
+from uqcsbot.models import Channel
+
+JOINED_PERMISSIONS = discord.Permissions(read_messages=True)
+SERVER_ID = 813324385179271168
+# Testing Server
+# SERVER_ID = 836589565237264415
+
+MESSAGE_ID = 950265978888007770
+
+EMOJIS = {"academic-advice": "📖", "adulting": "🧑", "banter": "💬", "bot-testing" : "🤖",
+                "contests" : "⚔️", "covid" : "⚛️", "creative" : "🎨", "events" : "🗓️", "food" : "🍔",
+                "games" : "🎮", "general" : "⚪", "hackathons" : "🍕", "hardware" : "💻", "jobs-bulletin" : "📌",
+                "jobs-discussion" : "🔈", "lgbtqia" : "🏳️‍🌈", "media" : "📺", "memes" : "🎭", "politics" : "📮",
+                "projects" : "🔨", "uqic-sport" : "🏅", "yelling" : "🗣️"}
+
+class Join(commands.Cog):
+
+    def __init__(self, bot: UQCSBot):
+        self.bot = bot
+
+    def _channel_query(self, channel: str):
+        db_session = self.bot.create_db_session()
+        channel_query = db_session.query(Channel).filter(Channel.name == channel,
+                                                         Channel.joinable == True).one_or_none()
+        db_session.close()
+        return channel_query
+
+    def get_key(self, map, value):
+        for k, v in map.items():
+            if v == value:
+                return k
+        return None
+
+    def get_channel_map(self):
+        db_session = self.bot.create_db_session()
+        channel_query = db_session.query(Channel).filter(Channel.joinable == True).order_by(Channel.name)
+        db_session.close()
+
+        channel_emojis = {}
+        for channel in channel_query:
+            if channel.name in EMOJIS:
+                channel_emojis[channel.name] = EMOJIS[channel.name]
+        return channel_emojis
+
+    @commands.Cog.listener()
+    async def on_raw_reaction_add(self, payload):
+        """ Toggle adding/removing member from the corresponding channel. """
+        channels = self.get_channel_map()
```

Comment:
this should probably only be done inside the if statement because this function is reasonably expensive and we dont want it to run every time a reaction is added